webui/src/pages/user/user-namespace-extension-list.tsx

```diff
@@ -0,0 +1,104 @@
+import { UserNamespaceExtensionListItem } from './user-namespace-extension-list-item';
+import { ExtensionRegistryService } from '../../extension-registry-service';
+import { ErrorResponse } from '../../server-request';
+import { DelayedLoadIndicator } from '../../custom-mui-components/delayed-load-indicator';
+
+const extensionListStyles = (theme: Theme) => createStyles({
+    container: {
+        padding: theme.spacing(5, 0),
+    },
+    extensions: {
+        display: 'grid',
+        gridTemplateColumns: `repeat(auto-fit, minmax(200px, 1fr))`,
+        gap: `1.5rem`,
+        marginTop: '1rem',
+    }
+});
+
+
+class UserNamespaceExtensionListComponent extends React.Component<UserNamespaceExtensionListComponent.Props, UserNamespaceExtensionListComponent.State> {
+
+    constructor(props: UserNamespaceExtensionListComponent.Props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+            extensions: undefined,
+        };
+    }
+
+    componentDidMount() {
+        this.initExtensions();
+    }
+
+    async initExtensions() {
+        const extensionsURLs: string[] = Object.keys(this.props.namespace.extensions).map((key: string) => this.props.namespace.extensions[key]);
+
+        const getExtension = async (url: string) => {
+            const extension = await this.props.service.getExtensionDetail(url);
+            return extension;
+        };
+
+        const getExtensions = async () => Promise.all(
+            extensionsURLs.map(
+                (url: string) => getExtension(url)
+            )
+        );
+
+        try {
+            const extensions = await getExtensions();
+            if (isError(extensions)) {
+                throw extensions;
+            } else {
+                this.setState({ extensions });
+            }
+        } catch (error) {
+            this.props.setError(error);
+        }
+    }
+
+    render() {
+        const { classes } = this.props;
+        return (
+            <section className={classes.container}>
+                <Typography variant="h5">Extensions</Typography>
+                <div className={classes.extensions}>
+                    {
+                        this.state.extensions ? this.state.extensions.map((extension: Extension, i: number) => <UserNamespaceExtensionListItem
+                            key={`${i}${extension.displayName}`}
+                            extension={extension}
+                        />) : <DelayedLoadIndicator loading={this.state.loading}/>
+                    }
+                </div>
+            </section>
+        );
+    }
+}
+
+export namespace UserNamespaceExtensionListComponent {
+    export interface Props extends WithStyles<typeof extensionListStyles> {
+        namespace: Namespace;
+        service: ExtensionRegistryService;
+        setError: (err: Error | Partial<ErrorResponse>) => void;
+    }
+
+    export interface State {
+        loading: boolean;
+        extensions: any;
```

Comment:
Remember to change this to `extensions?: Extension[]` after you've fixed the code above.